pkg/uroot/initramfs/archive.go

```diff
@@ -35,7 +36,7 @@ type Archiver interface {
 	//
 	// If `path` is unspecified, implementations may choose an arbitrary
 	// default location, potentially based on `goos` and `goarch`.
-	OpenWriter(path, goos, goarch string) (Writer, error)
+	OpenWriter(l logger.Logger, path, goos, goarch string) (Writer, error)
```

Comment:
I'd personally prefer giving the logger to the GetArchiver() command and have it construct an archiver that has the logger as an internal reference. That way if I implement the Archiver interface in my own software, I don't have to have whatever logging library I use implement the logger interface.